lib/block.mli

```diff
@@ -34,6 +34,21 @@ val blkgetsize: string -> Unix.file_descr -> [ `Ok of int64 | `Error of error ]
 
 val connect : string -> [`Ok of t | `Error of error] io
 
+type config = {
+  buffered: bool; (** true if I/O hits the OS disk caches, false if "direct" *)
+  sync: bool; (** true if [flush] flushes all caches, including disk drive caches *)
+  path: string; (** path to the underlying file *)
+}
+(** Configuration of a device *)
+
+val connect_uri : Uri.t -> [`Ok of t | `Error of error] io
```

Comment:
OK I've moved the `type config` into `type t` inside a `module Config` and added `to_string` and `of_string` as well as a convenient `make` function. I've removed `connect_uri` and added `connect_from_config` instead.